### PR TITLE
Run separate workflows to test monorepo-folders, nestjs-counter samples

### PR DESCRIPTION
**Pull request by vkarpov15**
## What was changed

Added a separate workflow to lint and test monorepo-folders and nestjs-counter because of the out of space issues with #169 .

## Why?

Tests started running out of disk space on Windows because installing a separate copy of the TypeScript SDK, along with all the other deps, for every single sample means that disk space use grows linearly with the number of samples. Eventually, with enough samples, we'll run out of disk space.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ name: CI
 
 on:
   push:
-    branches:
-      - main
```
Comment (lorensr) [inline .github/workflows/ci.yml:6]: Why run on push to all branches?

**Review by lorensr — APPROVED** [review]
Some lint warnings at bottom of Files changed tab

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vkarpov15) [reply .github/workflows/ci.yml:6]: I primarily made this change for testing, so the tests would actually run on my changes before I put in a PR. I'll revert it.

**Comment by vkarpov15** [comment]
The lint warnings are due to lack of decorator support, I'll look into figuring out how to add decorator support for eslint in the nestjs-counter example. Decorators are pretty important in Nest.

**Pull request merged**